### PR TITLE
feat(Blocks): Move blocks

### PR DESCRIPTION
**Pull request by madhurisandbhor**
### What does it do?

Adds the ability to move the blocks around in editor

### How to test it?

On hover user should be able to see drag option and able to drag block and list-items.

Note: There are some intermittent errors coming from slate and react-dnd, like "Expected targetIds to be registered" will be handled in coming PR. Workaround for now: you can try to drag item slowly to make it work.

### CODE REVIEW

File: packages/core/admin/admin/src/content-manager/hooks/useDragAndDrop.js
Comment (joshuaellis) [inline packages/core/admin/admin/src/content-manager/hooks/useDragAndDrop.js:1]: Can you explain the changes here?

**Review by joshuaellis — CHANGES_REQUESTED** [review]
I've not tested but just read the code and noted some areas of potential improvement. I think your PR description could be better, you've not just added drag and drop – you've also changed the universal hook `useDragAndDrop` for instance. So you should also add a why section even if one of the items is just "feature improvement" etc.

**Review by remidej — COMMENTED** [review]
there will be lots of conflicts to fix, but it's really cool to see this feature

you'll also need to add tests, probably E2E ones

File: packages/core/admin/admin/src/content-manager/components/BlocksEditor/BlocksInput/index.js [hunk 1]
```diff
@@ -17,6 +22,136 @@ const getEditorStyle = (theme) => ({
...
+  name,
+  onMoveItem,
+  blockType,
+}) => {
+  const [{ handlerId, isDragging, handleKeyDown }, myRef, boxRef, dropRef, dragRef] =
+    useDragAndDrop(!disabled && canDrag, {
+      type: `${ItemTypes.BLOCKS}._${name}`,
+      canDrop() {
+        return canDrag;
+      },
```
[the remark this answers is not in the document]
Comment (madhurisandbhor) [reply packages/core/admin/admin/src/content-manager/components/BlocksEditor/BlocksInput/index.js:85]: Here, I am trying to limit dnd over List block (not list items), even though you mark an item as not draggable, it still becomes droppable(using drop ref) as they are different sources, drag source and drop target.

**Comment by madhurisandbhor** [comment]
@joshuaellis @remidej Thank you for your valuable feedback 👍 , actually I didn't get time to review PR myself. I asked feedback a bit earlier as I was going to be away for a week. 
Also there are new updates on merging branches and typescript in place in blocks, I will create a new PR soon.

**Pull request closed without merging**